### ISSUE TITLE
Datman pipelines taking too long

### ISSUE THREAD

**Issue by tomwright01** [TIGRLab/admin#1113]
datman pipelines are taking around 19 hours to complete. dm_blacklist_rm.py is one culprit that could be relatively easy to refactor.

**Comment by DESm1th** [comment]
There's a lot of room to speed up blacklist_rm, but I think there must be another factor in the increase in run time. The pipelines ran fairly quickly until about 1-2 weeks ago and blacklist_rm has been running as part of the pipelines for far longer than that

**Comment by tomwright01** [comment]
Good point, pipelines are back to running in serial, no wonder the logs
were so neat. I've been clearing pipeline errors now. Assuming tomorrow
looks good I'll restore running in parallel.

**Comment by DESm1th** [comment]
Good catch! I just assumed the logs were neater because someone (I think Joe mentioned doing this) had written a script to concatenate the individual logs into a neat, orderly main log. Ha. I would never noticed they were serial again.

**Comment by josephdviviano** [comment]
@tomwright01 @DESm1th I think this is resolved, yes?

**Comment by DESm1th** [comment]
Not really. They still take about 24 hours to complete and are running serially.

**Comment by josephdviviano** [comment]
And we aren't running them in parallel just because of the logs?

**Comment by josephdviviano** [comment]
By 24 hours to complete, do you mean the queue jobs, or are you just saying that `blacklist_rm` and similar take too long

**Comment by DESm1th** [comment]
The entire nightly 'data' cron job takes ~24 hours to run in serial, sometimes a bit more or less depending on the number of new scans. 

Yeah, it was just done for the logs and was supposed to be temporary but I think we all forgot about this issue (I know I did). I think it might be good to switch to using the logging server so we can run in parallel and have neat logs, or else we should try using gnu parallel again since it buffers stdout + stderr and would also give us orderly logs. I almost got gnu parallel working waaay back last fall, but now I can't remember what the last problem was that needed to be resolved.
